internal/docstore/query.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2019 The Go Cloud Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package docstore
+
+import (
+	"context"
+	"io"
+
+	"gocloud.dev/internal/docstore/driver"
+)
+
+// Query represents a query over a collection.
+type Query struct {
+	coll *Collection
+	dq   *driver.Query
+}
+
+// Query creates a new Query over the collection.
+func (c *Collection) Query() *Query {
+	return &Query{coll: c, dq: &driver.Query{}}
+}
+
+// Where expresses a condition on the query.
+// Valid ops are: "=", ">", "<", ">=", "<=".
+func (q *Query) Where(field, op string, value interface{}) *Query {
+	q.dq.Filters = append(q.dq.Filters, driver.Filter{
+		Field: field,
+		Op:    op,
+		Value: value,
+	})
+	return q
+}
+
+// Limit will limit the results to at most n documents.
+func (q *Query) Limit(n int) *Query {
+	q.dq.Limit = n
+	return q
+}
+
+// Get returns an iterator for retrieving the documents specified by the query. If
+// field paths are provided, only those paths are set in the resulting documents.
+//
+// Call Stop on the iterator when finished.
```

Comment:
Why is `Stop` needed? Does it need to be called even if you iterate to the end?

WDYT about `Close` instead?

Reply:
Some providers may need to do resource cleanup if the user stops iterating in the middle. Firestore is one.

Naming a method `Close` without an error return is confusing to readers. Brad Fitzpatrick and I had a long discussion about this way back when I was writing the [iterator guidelines](https://github.com/googleapis/google-cloud-go/wiki/Iterator-Guidelines#the-close-and-stop-methods). I now agree with him: readers see a call to `Close` without an error check and get nervous. 

Why not `Close() error`? I don't believe any of the providers need to report an error back. That may change as we implement more.

Reply:
Some driver (firestore) needs to clean up the resource.

My understanding is that `Stop` doesn't return an error, like `Close` does, and doesn't need to be called if iterating to the end? Is that correct? @jba 

I'll add some documentation.